### PR TITLE
[docs] Add DeferNode and ConditionNode to query plan docs

### PR DESCRIPTION
**Pull request by smyrick**
Add docs for the different kinds of Nodes we can see in query plans

See full list in the code:
https://github.com/apollographql/federation/blob/9b7c6ee5405a48cb96e68d95736e6400903433b7/query-planner-js/src/QueryPlan.ts#L17

### CODE REVIEW

File: docs/source/query-plans.mdx [hunk 1]
```diff
@@ -131,10 +131,12 @@ Each node defined inside the `QueryPlan` node is one of the following:
...
-| [`Parallel`](#parallel-node) | Tells the gateway that the node's immediate children can be executed in parallel. |
-| [`Sequence`](#sequence-node) | Tells the gateway that the node's immediate children must be executed serially in the order listed. |
-| [`Flatten`](#flatten-node) | Tells the gateway to merge the data returned by this node's child `Fetch` node with data previously returned in the current `Sequence`. |
+| [`Fetch`](#fetch-node) | Tells the router to execute a particular operation on a particular subgraph. |
+| [`Parallel`](#parallel-node) | Tells the router that the node's immediate children can be executed in parallel. |
+| [`Sequence`](#sequence-node) | Tells the router that the node's immediate children must be executed serially in the order listed. |
+| [`Flatten`](#flatten-node) | Tells the router to merge the data returned by this node's child `Fetch` node with data previously returned in the current `Sequence`. |
+| [`Defer`](#defer-node) | Tells the router about one or more `@defer` blocks at the same nested level. It contains a primary block and an array of deffered blocks. |
```
Comment (Meschreiber) [inline docs/source/query-plans.mdx:138]:
```suggestion
| [`Defer`](#defer-node) | Tells the router about one or more blocks of `@defer`ed fields at the same level of nesting. The node contains a primary block and an array of deferred blocks. |
```

File: docs/source/query-plans.mdx [hunk 2]
```diff
@@ -444,6 +446,16 @@ You can view the query plan for a particular operation in any of the following w
...
+### Defer node
+A `Defer` node corresponds to one or more `@defer` at the same level of "nestedness" in the planned query.
+
+It contains a "primary block" and an array of "deferred blocks". The "primary block represents the part of the query that is not deferred (so the part of the query up until we reach the @defer(s) this handle), while each "deferred block" correspond to the deferred part of one of the `@defer` handle by the node.
+
+Read more about how `@defer` operatates at [Apollo Router support for @defer](/router/executing-operations/defer-support/).
+
+### Condition node
+A `Condition` node is used to split a plan into an if-else branch. This is commonly used for `@skip` and `@include` to have a query plan that can select different nodes based on the provided runtime variables.
+
```
Comment (Meschreiber) [inline docs/source/query-plans.mdx:458]: It looks like these need to move up to the previous H2, right under the `### Flatten node` section, so they are not part of `## Viewing query plans`

Additionally, could you please add examples of what the node looks like, as is done in the other node sections? E.g.  for [Flatten node](https://www.apollographql.com/docs/federation/query-plans/#flatten-node)

File: docs/source/query-plans.mdx [hunk 3]
```diff
@@ -444,6 +446,16 @@ You can view the query plan for a particular operation in any of the following w
     - Note that you must [register your federated graph](./managed-federation/setup/) with GraphOS to view query plans in the Explorer.
 - As direct output from the `@apollo/gateway` library (see below)
 
+### Defer node
+A `Defer` node corresponds to one or more `@defer` at the same level of "nestedness" in the planned query.
+
+It contains a "primary block" and an array of "deferred blocks". The "primary block represents the part of the query that is not deferred (so the part of the query up until we reach the @defer(s) this handle), while each "deferred block" correspond to the deferred part of one of the `@defer` handle by the node.
+
+Read more about how `@defer` operatates at [Apollo Router support for @defer](/router/executing-operations/defer-support/).
+
```
Comment (Meschreiber) [inline docs/source/query-plans.mdx:455]:
```suggestion

A `Defer` node corresponds to one or more `@defer`s at the same level of nesting in the query plan.

The node contains a _primary block_ and an array of _deferred blocks_. The primary block represents the part of the query that isn't deferred. Each deferred block corresponds to the one deferred part of the query.

Read more about how `@defer` operatates at [Apollo Router support for @defer](/router/executing-operations/defer-support/).

```

File: docs/source/query-plans.mdx [hunk 4]
```diff
@@ -444,6 +446,16 @@ You can view the query plan for a particular operation in any of the following w
...
+### Condition node
+A `Condition` node is used to split a plan into an if-else branch. This is commonly used for `@skip` and `@include` to have a query plan that can select different nodes based on the provided runtime variables.
```
Comment (Meschreiber) [inline docs/source/query-plans.mdx:457]:
```suggestion

A `Condition` node splits a query plan into an if-else branch. `Condition` nodes are commonly used for `@skip` and `@include` directives so the query plan to select different nodes based on the provided runtime variables.
```

File: docs/source/query-plans.mdx [hunk 5]
```diff
@@ -436,6 +438,59 @@ When the router sees this special `Fetch` syntax, it knows to query a subgraph's
...
+        }
+      }
+    ]
+  }
+}
+```
+
+### Condition node
+
+A `Condition` node splits a query plan into an if-else branch. `Condition` nodes are commonly used for `@skip` and `@include` directives so the query plan to select different nodes based on the provided runtime variables.
```
Comment (Meschreiber) [inline docs/source/query-plans.mdx:467]:
```suggestion
A `Condition` node splits a query plan into an if-else branch. `Condition` nodes are commonly used for `@skip` and `@include` directives so the query plan can select different nodes based on the provided runtime variables.
```

File: docs/source/query-plans.mdx [hunk 6]
```diff
@@ -436,6 +438,59 @@ When the router sees this special `Fetch` syntax, it knows to query a subgraph's
 
 Now that you've learned about each query plan node, take another look at the example query plan in [Example graph](#example-graph) to see how these nodes work together in a complete query plan.
 
+### Defer node
+
+A `Defer` node corresponds to one or more `@defer`s at the same level of nesting in the query plan.
+
+The node contains a _primary block_ and an array of _deferred blocks_. The primary block represents the part of the query that isn't deferred. Each deferred block corresponds to the one deferred part of the query.
+
+Read more about how `@defer` operatates at [Apollo Router support for @defer](/router/executing-operations/defer-support/).
```
Comment (Meschreiber) [inline docs/source/query-plans.mdx:447]:
```suggestion
Read more about how `@defer` works in the [Apollo Router support for @defer article](/router/executing-operations/defer-support/).
```

File: docs/source/query-plans.mdx [hunk 7]
```diff
@@ -436,6 +438,59 @@ When the router sees this special `Fetch` syntax, it knows to query a subgraph's
...
+QueryPlan {
+  Sequence {
+    Fetch(...) {}
+    Skip(...) {
+      Flatten(...) {
+        Fetch(...) { }
+      }
+    }
+  }
+}
```
Comment (Meschreiber) [inline docs/source/query-plans.mdx:491]: I don't see a `Condition` node in the current example. Is this correct?:

```suggestion
QueryPlan {
  Condition(if: $condition) {
    Then {
      Sequence {
        Fetch(...) {}
        Include(...) {
          Flatten(...) {
            Fetch(...) { }
          }
       }
    }  Else {
       Sequence {
         Fetch(...) {}
         Skip(...) {
           Flatten(...) {
             Fetch(...) { }
          }
       }
    }
 }
```

**Review by Meschreiber — APPROVED** [review]
Approved with a suggestion for an updated Query plan example for a `Condition` node

File: docs/source/query-plans.mdx (hunk 7 above)
Comment (smyrick) [reply docs/source/query-plans.mdx:491]: Yes, I will update the doc header. We renamed the node in a recent Federation version which I generated the example query plans from

**Pull request merged**